### PR TITLE
Salesforce crm dev

### PR DESCRIPTION
**Pull request by stevenlawrancesfdc**
Public Suffix List (PSL) Pull Request (PR) Template
====

Each PSL PR needs to have a description, rationale, indication of DNS validation and syntax checking, as well as a number of acknowledgements from the submitter.  This template must be included with each PR, and the submitting party MUST provide responses to all of the elements in order to be considered.

### Checklist of required steps

* [x] Description of Organization
* [x] Robust Reason for PSL Inclusion
* [x] DNS verification via dig
* [x] Run Syntax Checker (make test)

* [x] Each domain listed in the PRIVATE section has and shall maintain at least two years remaining on registration, and we shall keep the \_PSL txt record in place in the respective zone(s) in the affected section: Expiration is now 2029-02-12T23:46:32Z, as per https://whois-webform.markmonitor.com/whois/

__Submitter affirms the following:__ 

  * [x] We are listing *any* third-party limits that we seek to work around in our rationale such as those between IOS 14.5+ and Facebook (see [Issue #1245](https://github.com/publicsuffix/list/issues/1245) as a well-documented example)
We're not seeking to work around browser limits. Instead, we're seeking to have our entries treated as domain suffixes to maximize the fidelity of Salesforce's internal development environment with its production systems while making use of crm.dev instead of the actual production domains.

  * [x] This request was _not_ submitted with the objective of working around other third-party limits

  * [x] The [Guidelines](https://github.com/publicsuffix/list/wiki/Guidelines) were carefully _read_ and _understood_, and this request conforms
  * [x] The submission follows the [guidelines](https://github.com/publicsuffix/list/wiki/Format) on formatting and sorting

---

For Private section requests that are submitting entries for domains that match their organization website's primary domain, please understand that this can have impacts that may not match the desired outcome and take a long time to rollback, if at all.

To ensure that requested changes are entirely intentional, make sure that you read the affectation and propagation expectations, that you understand them, and confirm this understanding. 

PR Rollbacks have lower priority, and the volunteers are unable to control when or if browsers or other parties using the PSL will refresh or update.

(Link: [about propagation/expectations](https://github.com/publicsuffix/list/wiki/Guidelines#appropriate-expectations-on-derivative-propagation-use-or-inclusion))

 * [x] *Yes, I understand*.  I could break my organization's website cookies etc. and the rollback timing, etc is acceptable.  *Proceed*.
---

Description of Organization
====
Salesforce, Inc. is an American cloud-based software company headquartered in San Francisco, California. It provides customer relationship management (CRM) software and applications focused on sales, customer service, marketing automation, e-commerce, analytics, and application development. A core component of this is Salesforce's Sales, Service, and Experience Cloud, which is an integrated software system that undergoes continuous development by around 10,000 software developers.

I am Steven Lawrance, an architect on Salesforce's Sales, Service, and Experience Cloud that focuses on how the system makes use of DNS names for maximizing web security and isolation for customers in a scalable manner.

Organization Website: 

https://www.salesforce.com

Reason for PSL Inclusion
====
The Salesforce Sales, Service, and Experience Cloud system serves a large number of URL hostname patterns. To represent these in our internal development environment used by ~10,000 developers while separating the actual production and development domains, we use a URL hostname format that looks like MyDomainName.my.salesforce-com.WorkspaceID.w.crm.dev. To have WorkspaceID.w.crm.dev treated the same as 'com', we need to have *.w.crm.dev in the PSL. The salesforce-com (or force-com, etc.) subdomain thus gets treated as a registrable domain for cookie and privacy features within web browsers, mimicking the production salesforce.com and force.com domains. In addition to *.w.crm.dev, Salesforce has 26 partitions based on a letter that follows the 'w', hence the *.wa.crm.dev, *.wb.crm.dev, and so on names. Salesforce also has a separate development environment that uses 'd' instead of 'w', which is why *.d.crm.dev is in the list.

Number of users this request is being made to serve: Around 10,000 (estimated)

DNS Verification via dig
=======
```
dig +short TXT _psl.d.crm.dev
"https://github.com/publicsuffix/list/pull/1941"
```

```
dig +short TXT _psl.w.crm.dev
"https://github.com/publicsuffix/list/pull/1941"
```

```
dig +short TXT _psl.wa.crm.dev
"https://github.com/publicsuffix/list/pull/1941"
```

```
dig +short TXT _psl.wb.crm.dev
"https://github.com/publicsuffix/list/pull/1941"
```

```
dig +short TXT _psl.wc.crm.dev
"https://github.com/publicsuffix/list/pull/1941"
```

```
dig +short TXT _psl.wd.crm.dev
"https://github.com/publicsuffix/list/pull/1941"
```

```
dig +short TXT _psl.we.crm.dev
"https://github.com/publicsuffix/list/pull/1941"
```

```
dig +short TXT _psl.wf.crm.dev
"https://github.com/publicsuffix/list/pull/1941"
```

```
dig +short TXT _psl.wg.crm.dev
"https://github.com/publicsuffix/list/pull/1941"
```

```
dig +short TXT _psl.wh.crm.dev
"https://github.com/publicsuffix/list/pull/1941"
```

```
dig +short TXT _psl.wi.crm.dev
"https://github.com/publicsuffix/list/pull/1941"
```

```
dig +short TXT _psl.wj.crm.dev
"https://github.com/publicsuffix/list/pull/1941"
```

```
dig +short TXT _psl.wk.crm.dev
"https://github.com/publicsuffix/list/pull/1941"
```

```
dig +short TXT _psl.wl.crm.dev
"https://github.com/publicsuffix/list/pull/1941"
```

```
dig +short TXT _psl.wm.crm.dev
"https://github.com/publicsuffix/list/pull/1941"
```

```
dig +short TXT _psl.wn.crm.dev
"https://github.com/publicsuffix/list/pull/1941"
```

```
dig +short TXT _psl.wo.crm.dev
"https://github.com/publicsuffix/list/pull/1941"
```

```
dig +short TXT _psl.wp.crm.dev
"https://github.com/publicsuffix/list/pull/1941"
```

```
dig +short TXT _psl.wq.crm.dev
"https://github.com/publicsuffix/list/pull/1941"
```

```
dig +short TXT _psl.wr.crm.dev
"https://github.com/publicsuffix/list/pull/1941"
```

```
dig +short TXT _psl.ws.crm.dev
"https://github.com/publicsuffix/list/pull/1941"
```

```
dig +short TXT _psl.wt.crm.dev
"https://github.com/publicsuffix/list/pull/1941"
```

```
dig +short TXT _psl.wu.crm.dev
"https://github.com/publicsuffix/list/pull/1941"
```

```
dig +short TXT _psl.wv.crm.dev
"https://github.com/publicsuffix/list/pull/1941"
```

```
dig +short TXT _psl.ww.crm.dev
"https://github.com/publicsuffix/list/pull/1941"
```

```
dig +short TXT _psl.wx.crm.dev
"https://github.com/publicsuffix/list/pull/1941"
```

```
dig +short TXT _psl.wy.crm.dev
"https://github.com/publicsuffix/list/pull/1941"
```

```
dig +short TXT _psl.wz.crm.dev
"https://github.com/publicsuffix/list/pull/1941"
```

Results of Syntax Checker (`make test`)
=========

```
cd linter;                                \
	  ./pslint_selftest.sh;                     \
	  ./pslint.py ../public_suffix_list.dat;
-n test_NFKC: 
OK
-n test_allowedchars: 
OK
-n test_dots: 
OK
-n test_duplicate: 
OK
-n test_exception: 
OK
-n test_punycode: 
OK
-n test_section1: 
OK
-n test_section2: 
OK
-n test_section3: 
OK
-n test_section4: 
OK
-n test_spaces: 
OK
-n test_wildcard: 
OK
test -d libpsl || git clone --depth=1 https://github.com/rockdaboot/libpsl;   \
	  cd libpsl;                                                                    \
	  git pull;                                                                     \
	  echo "EXTRA_DIST =" >  gtk-doc.make;                                          \
	  echo "CLEANFILES =" >> gtk-doc.make;                                          \
	  autoreconf --install --force --symlink;
Already up to date.
autopoint: using AM_GNU_GETTEXT_REQUIRE_VERSION instead of AM_GNU_GETTEXT_VERSION
glibtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, 'build-aux'.
glibtoolize: linking file 'build-aux/ltmain.sh'
glibtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
glibtoolize: linking file 'm4/libtool.m4'
glibtoolize: linking file 'm4/ltoptions.m4'
glibtoolize: linking file 'm4/ltsugar.m4'
glibtoolize: linking file 'm4/ltversion.m4'
glibtoolize: linking file 'm4/lt~obsolete.m4'
configure.ac:1: warning: file 'version.txt' included several times
configure.ac:4: warning: file 'version.txt' included several times
aclocal.m4:765: AM_INIT_AUTOMAKE is expanded from...
configure.ac:4: the top level
configure.ac:369: warning: file 'version.txt' included several times
configure.ac:10: installing 'build-aux/compile'
configure.ac:4: installing 'build-aux/missing'
fuzz/Makefile.am: installing 'build-aux/depcomp'
cd libpsl && ./configure -q -C --enable-runtime=libicu --enable-builtin=libicu --with-psl-file=/Users/slawrance/dev/list/public_suffix_list.dat --with-psl-testfile=/Users/slawrance/dev/list/tests/tests.txt && make -s clean && make -s check -j4
configure: error: 'CPPFLAGS' was set to '-I/usr/local/opt/icu4c/include' in the previous run
configure: error: in '/Users/slawrance/dev/list/libpsl':
configure: error: changes in the environment can compromise the build
configure: error: run 'make distclean' and/or 'rm config.cache'
	    and start over
make: *** [libpsl-libicu] Error 1
slawran-ltm3b7r:list slawrance$ export CPPFLAGS=-I/usr/local/opt/icu4c/include
slawran-ltm3b7r:list slawrance$ make
cd linter;                                \
	  ./pslint_selftest.sh;                     \
	  ./pslint.py ../public_suffix_list.dat;
-n test_NFKC: 
OK
-n test_allowedchars: 
OK
-n test_dots: 
OK
-n test_duplicate: 
OK
-n test_exception: 
OK
-n test_punycode: 
OK
-n test_section1: 
OK
-n test_section2: 
OK
-n test_section3: 
OK
-n test_section4: 
OK
-n test_spaces: 
OK
-n test_wildcard: 
OK
test -d libpsl || git clone --depth=1 https://github.com/rockdaboot/libpsl;   \
	  cd libpsl;                                                                    \
	  git pull;                                                                     \
	  echo "EXTRA_DIST =" >  gtk-doc.make;                                          \
	  echo "CLEANFILES =" >> gtk-doc.make;                                          \
	  autoreconf --install --force --symlink;
Already up to date.
autopoint: using AM_GNU_GETTEXT_REQUIRE_VERSION instead of AM_GNU_GETTEXT_VERSION
glibtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, 'build-aux'.
glibtoolize: linking file 'build-aux/ltmain.sh'
glibtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
glibtoolize: linking file 'm4/libtool.m4'
glibtoolize: linking file 'm4/ltoptions.m4'
glibtoolize: linking file 'm4/ltsugar.m4'
glibtoolize: linking file 'm4/ltversion.m4'
glibtoolize: linking file 'm4/lt~obsolete.m4'
configure.ac:1: warning: file 'version.txt' included several times
configure.ac:4: warning: file 'version.txt' included several times
aclocal.m4:765: AM_INIT_AUTOMAKE is expanded from...
configure.ac:4: the top level
configure.ac:369: warning: file 'version.txt' included several times
configure.ac:10: installing 'build-aux/compile'
configure.ac:4: installing 'build-aux/missing'
fuzz/Makefile.am: installing 'build-aux/depcomp'
cd libpsl && ./configure -q -C --enable-runtime=libicu --enable-builtin=libicu --with-psl-file=/Users/slawrance/dev/list/public_suffix_list.dat --with-psl-testfile=/Users/slawrance/dev/list/tests/tests.txt && make -s clean && make -s check -j4
configure: WARNING: --enable-builtin=libicu is deprecated, use --enable-builtin (enabled by default)
config.status: creating po/POTFILES
config.status: creating po/Makefile
Making clean in po
Making clean in include
Making clean in src
rm -f ./so_locations
Making clean in tools
 rm -f psl
Making clean in fuzz
 rm -f libpsl_icu_fuzzer libpsl_icu_load_fuzzer libpsl_icu_load_dafsa_fuzzer
Making clean in tests
 rm -f test-is-public test-is-public-all test-is-cookie-domain-acceptable test-is-public-builtin test-registrable-domain
Making clean in msvc
Making check in po
Making check in include
Making check in src
  CC       libpsl_la-psl.lo
  CC       libpsl_la-lookup_string_in_fixed_set.lo
  CCLD     libpsl.la
Making check in tools
  CC       psl.o
  CCLD     psl
Making check in fuzz
  CC       libpsl_fuzzer.o
  CC       main.o
  CC       libpsl_load_fuzzer.o
  CC       libpsl_load_dafsa_fuzzer.o
  CCLD     libpsl_icu_fuzzer
  CCLD     libpsl_icu_load_fuzzer
  CCLD     libpsl_icu_load_dafsa_fuzzer
PASS: libpsl_icu_load_dafsa_fuzzer
PASS: libpsl_icu_fuzzer
PASS: libpsl_icu_load_fuzzer
============================================================================
Testsuite summary for libpsl 0.21.5
============================================================================
# TOTAL: 3
# PASS:  3
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
Making check in tests
  CC       test-is-public.o
  CC       test-is-cookie-domain-acceptable.o
  CC       common.o
  CC       test-is-public-all.o
  CC       test-is-public-builtin.o
  CC       test-registrable-domain.o
  CCLD     test-is-public
  CCLD     test-is-cookie-domain-acceptable
  CCLD     test-is-public-all
  CCLD     test-is-public-builtin
libtool: warning: '-no-install' is ignored for x86_64-apple-darwin23.3.0
libtool: warning: '-no-install' is ignored for x86_64-apple-darwin23.3.0
libtool: warning: '-no-install' is ignored for x86_64-apple-darwin23.3.0
libtool: warning: '-no-install' is ignored for x86_64-apple-darwin23.3.0
libtool: warning: assuming '-no-fast-install' instead
libtool: warning: assuming '-no-fast-install' instead
libtool: warning: assuming '-no-fast-install' instead
libtool: warning: assuming '-no-fast-install' instead
  CCLD     test-registrable-domain
libtool: warning: '-no-install' is ignored for x86_64-apple-darwin23.3.0
libtool: warning: assuming '-no-fast-install' instead
PASS: test-is-public
PASS: test-is-public-builtin
PASS: test-is-cookie-domain-acceptable
PASS: test-is-public-all
PASS: test-registrable-domain
============================================================================
Testsuite summary for libpsl 0.21.5
============================================================================
# TOTAL: 5
# PASS:  5
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
Making check in msvc
```

### CODE REVIEW

**Comment by stevenlawrancesfdc** [comment]
Good point. I'll extend the existing Salesforce.com block.

**Comment by stevenlawrancesfdc** [comment]
@yahesh @simon-friedberger Thanks! I now have the two Salesforce sections merged together.

**Comment by yahesh** [comment]
LGTM.

File: public_suffix_list.dat [hunk 1]
```diff
@@ -14860,11 +14860,39 @@ from.tv
 sakura.tv
 
 // Salesforce.com, Inc. https://salesforce.com/
-// Submitted by Michael Biven <mbiven@salesforce.com> and Aaron Romeo <aaron.romeo@salesforce.com>
+// Submitted by Michael Biven <mbiven@salesforce.com>, Aaron Romeo <aaron.romeo@salesforce.com>, and Steven Lawrance <slawrance@salesforce.com>
```
Comment (simon-friedberger) [inline public_suffix_list.dat:14863]: Please replace this with a single non-personal e-mail.

File: public_suffix_list.dat [hunk 2]
```diff
@@ -14860,11 +14860,39 @@ from.tv
...
+*.wq.crm.dev
+*.wr.crm.dev
+*.ws.crm.dev
+*.wt.crm.dev
+*.wu.crm.dev
+*.wv.crm.dev
+*.ww.crm.dev
+*.wx.crm.dev
+*.wy.crm.dev
+*.wz.crm.dev
```
Comment (simon-friedberger) [inline public_suffix_list.dat:14895]: Is it possible to compress this further? Are all these individual domains necessary?

Comment (simon-friedberger) [reply public_suffix_list.dat:14895]: Users-per-line might be too low for acceptance like this.

**Comment by simon-friedberger** [comment]
ping @stevenlawrancesfdc

**Comment by stevenlawrancesfdc** [comment]
@simon-friedberger Thanks, and sorry for my delay here. I'll apply the feedback to this pull request now.

File: public_suffix_list.dat (hunk 1 above)
Comment (stevenlawrancesfdc) [reply public_suffix_list.dat]: @simon-friedberger Thanks I now have that done in the newest commit, where those colleagues and I are collapsed into public-suffix-list@salesforce.com.

File: public_suffix_list.dat (hunk 2 above)
Comment (stevenlawrancesfdc) [reply public_suffix_list.dat]: I now have the list cut down to what is presently in operation plus the upcoming ones.

Thanks!

**Comment by simon-friedberger** [comment]
* [x] Expiration (Note: Must STAY >2y at all times)
  * [x] crm.dev expires 2029-02-12
* [x] DNS _psl entries (Note: Must STAY in place)
* [x] Tests pass
* [x] Sorting
* [x] Reasoning/Organization description

**Pull request merged**